src/utils/plotSpecs/generateEmbeddingCategoricalSpec.js

```diff
@@ -184,7 +184,7 @@ const generateSpec = (config, plotData) => {
 };
 
 const filterCells = (cellSets, selectedCellSet) => {
-  let newCellSets = cellSets.hierarchy.find(
+  let newCellSets = cellSets.hierarchy?.find(
```

Comment:
This is probably not the place to do this, as there is no guarantee that `filterCells` and `generateData` will re-run once the cell sets become available.

Are you sure this is a problem at all? Can you tell me how you managed to produce the bug? I checked `CategoricalEmbeddingPlot` and it seems that there is a check to ensure this plot does not load unless cell sets are available (see [here](https://github.com/biomage-ltd/ui/blob/master/src/components/plots/CategoricalEmbeddingPlot.jsx#L47-L51)).

Reply:
I was looking there also, until I realised that the issue came from https://github.com/biomage-ltd/ui/blob/57d9e8df13e877690e49f9ac06b4598651efb1aa/src/components/data-processing/ConfigureEmbedding/ConfigureEmbedding.jsx#L336